openmc/macroscopic.py

```diff
@@ -0,0 +1,85 @@
+from numbers import Integral
+import sys
+
+from openmc.checkvalue import check_type
+
+if sys.version_info[0] >= 3:
+    basestring = str
+
+
+class Macroscopic(object):
```

Comment:
In XML input, calling an element of a material "macroscopic" (instead of a nuclide or element) means you are giving a macroscopic cross section instead of microscopic (and thus there is no need to deal with number densities etc).  This quantity class therefore mimics this by duplicating the Nuclide class.  I could have added a flag to Nuclide to be like "is_macroscopic" but this seemed less awkward.